x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/components/templates_form/templates_form.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useEffect, useRef } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { EuiText, EuiLink, EuiSpacer } from '@elastic/eui';
+import { useForm, Form, SerializerFunc, UseField, JsonEditorField } from '../../shared_imports';
+import { Types, useDispatch } from '../../mappings_state';
+import { templatesFormSchema } from './templates_form_schema';
+import { documentationService } from '../../../../services/documentation';
+
+type MappingsTemplates = Types['MappingsTemplates'];
+
+interface Props {
+  defaultValue?: MappingsTemplates;
+}
+
+const stringifyJson = (json: { [key: string]: any }) =>
+  Array.isArray(json) ? JSON.stringify(json, null, 2) : '[\n\n]';
+
+const formSerializer: SerializerFunc<MappingsTemplates> = formData => {
+  const { dynamicTemplates } = formData;
+
+  let parsedTemplates;
+  try {
+    parsedTemplates = JSON.parse(dynamicTemplates);
```

Comment:
I realized that if we pass just an object instead of an array of objects, we lose our dynamic template.

Can we add this?

```js
parsedTemplates = JSON.parse(dynamicTemplates);
if (!Array.isArray(parsedTemplates)) {
  // It means that it is an object, but we need an array of objects
  parsedTemplates = [parsedTemplates];
}
```

Reply:
good catch, fixed!